src/arch/riscv/irqc/aia/inc/vaplic.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Bao Project and Contributors. All rights reserved.
+ */
+
+#ifndef __VAPLIC_H__
+#define __VAPLIC_H__
+
+#include <bao.h>
+#include <aplic.h>
+#include <arch/spinlock.h>
+#include <bitmap.h>
+#include <emul.h>
+
+/**
+ * @brief   Data structure to the virtual interrupt controller when a APLIC 
+ *          is the platform interrupt controller.
+ * 
+ */
+struct virqc {
+    spinlock_t lock;
+    size_t idc_num;
+    BITMAP_ALLOC(hw, APLIC_MAX_INTERRUPTS);
+    uint32_t domaincfg;
+    uint32_t srccfg[APLIC_MAX_INTERRUPTS-1];
+    uint32_t setip[APLIC_MAX_INTERRUPTS/32];
+    uint32_t in_clrip[APLIC_MAX_INTERRUPTS/32];
+    uint32_t setie[APLIC_MAX_INTERRUPTS/32];
+    uint32_t target[APLIC_MAX_INTERRUPTS-1];
+    BITMAP_ALLOC(idelivery, APLIC_PLAT_IDC_NUM);
+    BITMAP_ALLOC(iforce, APLIC_PLAT_IDC_NUM);
+    uint32_t ithreshold[APLIC_PLAT_IDC_NUM];
+    uint32_t topi_claimi[APLIC_PLAT_IDC_NUM];
+    struct emul_mem aplic_domain_emul;
+    struct emul_mem aplic_idc_emul;
+};
+
+struct vm;
+struct vcpu;
+
+/**
+ * @brief Initialize the virtual APLIC for a given virtual machine.
+ * 
+ * @param vm Virtual machine associated to the virtual APLIC
+ * @param vaplic_base address base of the physical APLIC
+ * 
+ */
+void vxplic_init(struct vm *vm, struct arch_platform *arch_platform);
```

Comment:
I'd prefer the second parameter would be something a bit more interrupt controller specific. Maybe we could add an anonymous struct fot that in the riscv arch_platform struct.

Reply:
Okay, I can send the irqc union as the second parameter. Or, wrap it in a struct and pass this struct.